src/pivot-table/hooks/use-column-width.ts

```diff
@@ -142,89 +137,105 @@ export default function useColumnWidth(
-          break;
-        }
-        case ColumnWidthType.FitToContent: {
-          specifiedWidth = topGridLeavesIsPseudo
-            ? Math.max(estimateWidthForContent(qApprMaxGlyphCount), measureTextForColumnContent(qFallbackTitle))
-            : Math.max(
-                Math.max(...qMeasureInfo.map((m) => estimateWidthForContent(m.qApprMaxGlyphCount))),
-                estimateWidthForColumnContent(qApprMaxGlyphCount) + leavesIconWidth,
-              );
-          break;
-        }
-        case ColumnWidthType.Auto: {
-          // TODO: we might need to redo this in the pseudo dimension case, since if not all measure are set to auto
-          // you don't fill upp the width anyway
-          specifiedWidth = autoWidth;
-          break;
+        const addKnownWidth = (width: number) => {
+          widths[idx] = Math.min(ColumnWidthValues.PixelsMax, Math.max(ColumnWidthValues.PixelsMin, width));
+          // remove the width * number of instances of that width from the remaining width for auto columns
+          sumAutoWidths -= widths[idx] * (size.x / columnArray.length);
+        };
+
+        switch (type) {
+          case ColumnWidthType.Pixels:
+            addKnownWidth(getPixelValue(pixels));
+            break;
+          case ColumnWidthType.Percentage:
+            addKnownWidth(getPercentageValue(percentage) * rightGridAvailableWidth);
+            break;
+          case ColumnWidthType.FitToContent:
+            // eslint-disable-next-line no-case-declarations
+            const fitToContentWidth = topGridLeavesIsPseudo
+              ? Math.max(
+                  estimateWidthForContent(qApprMaxGlyphCount),
+                  measureTextForColumnContent(qFallbackTitle) + TOTAL_CELL_PADDING,
+                )
+              : Math.max(
+                  Math.max(...qMeasureInfo.map((m) => estimateWidthForContent(m.qApprMaxGlyphCount))),
+                  estimateWidthForColumnContent(qApprMaxGlyphCount) + leavesIconWidth,
+                );
+            addKnownWidth(fitToContentWidth);
+            break;
+          case ColumnWidthType.Auto:
+          default:
+            // stores the indexes of auto columns to loop over later
+            autoColumnIndexes.push(idx);
+            break;
         }
-        default:
-          specifiedWidth = autoWidth;
-          break;
+      } else {
+        autoColumnIndexes.push(idx);
       }
+    });
 
-      return Math.max(ColumnWidthValues.PixelsMin, specifiedWidth);
-    },
-    [
-      rightGridAvailableWidth,
-      layoutService.size.x,
-      topGridLeavesIsPseudo,
-      estimateWidthForContent,
-      measureTextForColumnContent,
-      qMeasureInfo,
-      estimateWidthForColumnContent,
-      leavesIconWidth,
-    ],
-  );
+    if (autoColumnIndexes.length) {
+      // divides remaining width evenly between auto columns
+      const totalNumberOfAutoColumns = (size.x * autoColumnIndexes.length) / columnArray.length;
```

Comment:
shouldn't `totalNumberOfAutoColumns` be equal to just `autoColumnIndexes.length`? 
otherwise the name is misleading in this case, because what this variable store is something else

Reply:
yeah naming could probably be better here. The thing is that we loop over the number of measure, let's say there are 3 => `columnArray.length === 3`. Then one of them is  non-auto => `autoColumnIndexes.length === 2`. Now there could be multiple instances of each measure, if you have a dimension above it for example, let's say a dimension with 10 values => `size.x === 30`. So what `totalNumberOfAutoColumns` is, is the actual number of columns that will have the auto width => 30 * 2 / 3 = 20.

So maybe the name could be `autoColumnInstances` or something?

Reply:
Now i understand the logic, right!
Yes, the new name is great. 
Also by that, it would be nice to change `autoWidth` (in the line below) to something like `possibleAutoColWidth` or `tunedAutoColWidth` or something better if you have in mind. 

Also can you add this example in code as well, I think it would really help us understand the code better later on when we come back to code base

Reply:
sure I can comment something

Reply:
I think `autoWidth` is good, since that is the width every auto column would get (with the exception of when it is lower than the min value)